guide/docs/popular-topics/permissions.mdx

```diff
@@ -1,8 +1,382 @@
+    await inter.send("Sure you can!" if inter.permissions.add_reactions else "Sadly no :(")
+```
+
+## Querying permissions
+
+There are various ways to query permission information, and they also depend on the context you're
+in. This section tries to list all possible methods for getting permission info, grouped by context
+they can be used in.
+
+### Context-agnostic
+
+If you have an instance of any class implementing <DocsLink reference="disnake.abc.GuildChannel" />,
+then the easiest way to query permissions for a member/role is
+through <DocsLink reference="disnake.abc.GuildChannel.permissions_for" />.
+If you need a bit more lower level control, you can also get raw overwrite objects
+using <DocsLink reference="disnake.abc.GuildChannel.overwrites_for" />
+or <DocsLink reference="disnake.abc.GuildChannel.overwrites" />.
+
+### In prefix commands
+
+There are no direct methods on <DocsLink reference="disnake.ext.commands.Context" /> for querying permission
+information, however, you can still access <DocsLink reference="disnake.ext.commands.Context.channel" /> and
+call above-mentioned methods on it, though be sure to check that it's not <DocsLink reference="disnake.PartialMessageable" />
+as it doesn't have any permissions-related methods on it either.
+
+### In interactions
+
+When working with interactions (i.e., with application commands or components), there are two main methods
+defined on <DocsLink reference="disnake.Interaction" />: <DocsLink reference="disnake.Interaction.permissions" />
+(for getting interaction author's permissions) and <DocsLink reference="disnake.Interaction.app_permissions" />
+(for getting bot's permissions in interaction's channel).
+
+## How permissions are calculated
+
+Permissions can be a tough thing to understand at first, but understanding how permissions are
+applied can help you <!-- feel the pain of us mere mortals re-implementing discord client's obscure behaviour -->
+better grasp this concept.
+
+In case we're calculating permissions for a member, the strategy is as follows:
+
+-   If member is the guild owner, all permissions are granted - no questions asked. Otherwise..
+-   First we apply the `@everyone` role's permissions.
+-   Then, for each role the member has we apply it's permissions too.
+-   -   If at this point the member has the "Administrator" permission, all others are granted automatically
+        (the member is a guild-wide administrator, and, as such, is functionally equivalent to the owner except
+        they can't delete the guild).
+-   After that, if we're in a channel, the following is additionally done.
+-   -   We apply it's overwrites in the same order: `@everyone` overwrites,
+        role overwrites and finally member overwrites.
+-   -   Then [channel-specific implicit permissions](#channel-specific-implicit-permissions) are handled.
+-   In the end, we take into account [guild-wide implicit permissions](#guild-wide-implicit-permissions).
```

Comment:
this should probably be an enumeration

Reply:
a15b78b18fc6f63696fa98d4ed842b5095386688